pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentPushUtils.java

```diff
@@ -134,6 +135,8 @@ public static void pushSegments(SegmentGenerationJobSpec spec, PinotFS fileSyste
                       segmentName, controllerURI, e);
               throw e;
             }
+          } finally {
+            FileUtils.deleteQuietly(tarFile);
```

Comment:
Please use flag to turn on/off this deletion.

Reply:
Added `_cleanUpOutputDir` in jobSpec and defaults to `false`. Have tested for jobType `SegmentCreationAndTarPush` and `SegmentCreationAndUriPush` with `_cleanUpOutputDir=true/false`